lib/segment/src/index/field_index/full_text_index/mutable_inverted_index.rs

```diff
@@ -109,27 +115,63 @@ impl InvertedIndex for MutableInvertedIndex {
 
             hw_cell_wb.incr_delta(size_of_val(&point_id));
         }
-        self.point_to_docs[point_id as usize] = Some(document);
+        self.point_to_tokens[point_id as usize] = Some(tokens);
+
+        Ok(())
+    }
+
+    fn index_document(
+        &mut self,
+        point_id: PointOffsetType,
+        ordered_document: Document,
+        hw_counter: &HardwareCounterCell,
+    ) -> OperationResult<()> {
+        let Some(point_to_doc) = &mut self.point_to_doc else {
+            // Phrase matching is not enabled
+            return Ok(());
+        };
+
+        let mut hw_cell_wb = hw_counter
+            .payload_index_io_write_counter()
+            .write_back_counter();
+
+        // Ensure container has enough capacity
+        if point_to_doc.len() <= point_id as usize {
+            let new_len = point_id as usize + 1;
+
+            hw_cell_wb.incr_delta((new_len - point_to_doc.len()) * size_of::<Option<Document>>());
```

Comment:
maybe we could batch all measurements into a single call

Reply:
the WritebackCounterCell already handles this logic internally 👍